Import/Importer.cpp

```diff
@@ -549,6 +552,275 @@ void TypedImportBuffer::pop_value() {
+}
+
+static void appendArrowTime(const ColumnDescriptor* cd, const Array& values, std::vector<time_t>* buffer) {
+  const time_t null_sentinel = inline_fixed_encoding_null_val(cd->columnType);
+  if (values.type_id() == Type::TIME32) {
+    const auto& typed_values = static_cast<const Time32Array&>(values);
+    const auto& type = static_cast<const Time32Type&>(*values.type());
+
+    buffer->reserve(typed_values.length());
+    const int32_t* raw_values = typed_values.raw_values();
+    const TimeUnit::type unit = type.unit();
+
+    for (int64_t i = 0; i < typed_values.length(); i++) {
+      if (typed_values.IsNull(i)) {
+        buffer->push_back(null_sentinel);
+      } else {
+        switch (unit) {
+          case TimeUnit::SECOND:
+            buffer->push_back(static_cast<time_t>(raw_values[i]));
+            break;
+          case TimeUnit::MILLI:
+            buffer->push_back(static_cast<time_t>(raw_values[i] / 1000));
+            break;
+          default:
+            // unreachable code
+            CHECK(false);
+            break;
+        }
+      }
+    }
+  } else if (values.type_id() == Type::TIME64) {
+    const auto& typed_values = static_cast<const Time64Array&>(values);
+    const auto& type = static_cast<const Time64Type&>(*values.type());
+
+    buffer->reserve(typed_values.length());
+    const int64_t* raw_values = typed_values.raw_values();
+    const TimeUnit::type unit = type.unit();
+
+    for (int64_t i = 0; i < typed_values.length(); i++) {
+      if (typed_values.IsNull(i)) {
+        buffer->push_back(null_sentinel);
+      } else {
+        switch (unit) {
+          case TimeUnit::MICRO:
+            buffer->push_back(static_cast<time_t>(raw_values[i] / 1000000L));
+            break;
+          case TimeUnit::NANO:
+            buffer->push_back(static_cast<time_t>(raw_values[i] / 1000000000L));
+            break;
+          default:
+            // unreachable code
+            CHECK(false);
+            break;
+        }
+      }
+    }
+  } else {
+    ARROW_THROW_IF(true, "Column was not time32 or time64");
+  }
+}
+
+static void appendArrowTimestamp(const ColumnDescriptor* cd, const Array& values, std::vector<time_t>* buffer) {
+  ARROW_THROW_IF(values.type_id() != Type::TIMESTAMP, "Expected timestamp col");
+
+  const time_t null_sentinel = inline_fixed_encoding_null_val(cd->columnType);
+  const auto& typed_values = static_cast<const TimestampArray&>(values);
+  const auto& type = static_cast<const TimestampType&>(*values.type());
+
+  buffer->reserve(typed_values.length());
+  const int64_t* raw_values = typed_values.raw_values();
+  const TimeUnit::type unit = type.unit();
+
+  for (int64_t i = 0; i < typed_values.length(); i++) {
+    if (typed_values.IsNull(i)) {
+      buffer->push_back(null_sentinel);
+    } else {
+      switch (unit) {
+        case TimeUnit::SECOND:
+          buffer->push_back(static_cast<time_t>(raw_values[i]));
+          break;
+        case TimeUnit::MILLI:
```

Comment:
These conversions make me slightly nervous, I had to count the zeros. Using `std::chrono::time_point` feels very heavy-handed, maybe write 1B like 1000 * 1000 * 1000 or something? At least until we require a C++14 compiler, then we can use digit separators.

Reply:
done